Makefile

```diff
@@ -52,6 +53,9 @@ JAVAC_OPTIONS=--release $(JAVA_TARGET) -Xlint:-options
 
 TEST_LIB_DIR=build/test/lib
 TEST_BIN_DIR=build/test/bin
+TEST_DEPS_DIR=test/deps
+TEST_GEN_DIR=test/gen
+TEST_STUBS_DIR=test/stub-jars
```

Comment:
Maybe put it in `test/gen`? Why another directory?
I'd also rename file to `protobuf-stubs.jar` to make it clear it has nothing to do with the original `protobuf-java`, i.e., it's not a derived work and is not subject to the protobuf license T&C.

Reply:
Mixing stubs with other dirs makes it hard to reason about the jar order in the classpath. As it is now, we now for sure that stubs will always appear in the rightmost part of the classpath definition.